RoaringBitmap.andCardinality is not reflexive with run containers

RoaringBitmap.andCardinality should return the cardinality of the bitmap when the two bitmaps are identical. RoaringBitmap.equals respects this relation, but RoaringBitmap.andCardinality fails when there are run containers in the bitmap.

The second assertion below fails, whereas the first does not.

   ```
@Test
    public void testAndCardinality() {
        RoaringBitmap baseline = new RoaringBitmap();
        baseline.add((1L << 16) + 1, 1L << 18);
        baseline.add((1L << 19) + 1, 1L << 20);
        baseline.add((1 << 21) + 1);
        baseline.add((1 << 21) + 3);
        baseline.add((1 << 21) + 5);
        assertEquals(baseline, RoaringBitmap.and(baseline, baseline));
        assertEquals(baseline.getCardinality(), RoaringBitmap.andCardinality(baseline, baseline));
    }
```

java.lang.AssertionError: 
Expected :720897
Actual   :1

--- a/src/main/java/org/roaringbitmap/RunContainer.java
+++ b/src/main/java/org/roaringbitmap/RunContainer.java
@@ -560,7 +560,7 @@ public final class RunContainer extends Container implements Cloneable {
           }
         }
         // earliestend - lateststart are all values that are true.
-        cardinality += (short) (earliestend - lateststart);
+        cardinality += earliestend - lateststart;
       }
     }
     return cardinality;

--- a/src/main/java/org/roaringbitmap/buffer/MappeableRunContainer.java
+++ b/src/main/java/org/roaringbitmap/buffer/MappeableRunContainer.java
@@ -2482,7 +2482,7 @@ public final class MappeableRunContainer extends MappeableContainer implements C
           }
         }
         // earliestend - lateststart are all values that are true.
-        cardinality += (short) (earliestend - lateststart);
+        cardinality += earliestend - lateststart;
       }
     }
     return cardinality;

--- a/src/test/java/org/roaringbitmap/TestRoaringBitmap.java
+++ b/src/test/java/org/roaringbitmap/TestRoaringBitmap.java
@@ -4440,4 +4440,16 @@ public class TestRoaringBitmap {
     Assert.assertEquals((1L << 30) - 1, rb.last());
   }
 
+  @Test
+  public void testAndCardinality() {
+    RoaringBitmap baseline = new RoaringBitmap();
+    baseline.add((1L << 16) + 1, 1L << 18);
+    baseline.add((1L << 19) + 1, 1L << 20);
+    baseline.add((1 << 21) + 1);
+    baseline.add((1 << 21) + 3);
+    baseline.add((1 << 21) + 5);
+    assertEquals(baseline, RoaringBitmap.and(baseline, baseline));
+    assertEquals(baseline.getCardinality(), RoaringBitmap.andCardinality(baseline, baseline));
+  }
+
 }

--- a/src/test/java/org/roaringbitmap/buffer/TestRoaringBitmap.java
+++ b/src/test/java/org/roaringbitmap/buffer/TestRoaringBitmap.java
@@ -3471,5 +3471,17 @@ public class TestRoaringBitmap {
     Assert.assertEquals((1L << 30) - 1, rb.last());
   }
 
+  @Test
+  public void testAndCardinality() {
+    MutableRoaringBitmap baseline = new MutableRoaringBitmap();
+    baseline.add((1L << 16) + 1, 1L << 18);
+    baseline.add((1L << 19) + 1, 1L << 20);
+    baseline.add((1 << 21) + 1);
+    baseline.add((1 << 21) + 3);
+    baseline.add((1 << 21) + 5);
+    assertEquals(baseline, MutableRoaringBitmap.and(baseline, baseline));
+    assertEquals(baseline.getCardinality(), MutableRoaringBitmap.andCardinality(baseline, baseline));
+  }
+
 
 }

